Build failure: g810-led

### Steps To Reproduce

Steps to reproduce the behavior:
1. build g810-led

### Build log

```
error: builder for '/nix/store/g5dyv8f7shq54cjj6xsw102q654r59yl-g810-led-0.4.3.drv' failed with exit code 2;
       last 10 log lines:
       > src/helpers/help.cpp:353:32: error: 'KeyboardFeatures' is not a class, namespace, or enumeration
       >   353 |                 if((features | KeyboardFeatures::intensity) == features) {
       >       |                                ^~~~~~~~~~~~~~~~
       > src/helpers/help.cpp:360:21: error: 'features' was not declared in this scope
       >   360 |                 if((features | KeyboardFeatures::setkey) == features) {
       >       |                     ^~~~~~~~
       > src/helpers/help.cpp:360:32: error: 'KeyboardFeatures' is not a class, namespace, or enumeration
       >   360 |                 if((features | KeyboardFeatures::setkey) == features) {
       >       |                                ^~~~~~~~~~~~~~~~
       > make: *** [makefile:35: bin/g810-led] Error 1
       For full logs, run 'nix log /nix/store/g5dyv8f7shq54cjj6xsw102q654r59yl-g810-led-0.4.3.drv'.
```

### Notify maintainers

@fabaff 

### Metadata

Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
this path will be fetched (0.00 MiB download, 0.00 MiB unpacked):
  /nix/store/jlx0nh7j7lfsi1sfjlvsp7f0bfjgnnaf-nix-info
copying path '/nix/store/jlx0nh7j7lfsi1sfjlvsp7f0bfjgnnaf-nix-info' from 'https://cache.nixos.org'...
 - system: `"x86_64-linux"`
 - host os: `Linux 6.1.71, NixOS, 24.05 (Uakari), 24.05.20240108.317484b`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.18.1`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/misc/g810-led/default.nix
+++ b/pkgs/misc/g810-led/default.nix
@@ -5,14 +5,14 @@
 , profile ? "/etc/g810-led/profile"
 }:
 
-stdenv.mkDerivation rec {
+stdenv.mkDerivation (finalAttrs: {
   pname = "g810-led";
   version = "0.4.3";
 
   src = fetchFromGitHub {
     owner = "MatMoul";
-    repo = pname;
-    rev = "refs/tags/v${version}";
+    repo = "g810-led";
+    rev = "refs/tags/v${finalAttrs.version}";
     hash = "sha256-GKHtQ7DinqfhclDdPO94KtTLQhhonAoWS4VOvs6CMhY=";
   };
 
@@ -21,6 +21,9 @@ stdenv.mkDerivation rec {
       --replace "/usr" $out \
       --replace "/etc/g810-led/profile" "${profile}"
   '';
+
+  # GCC 13 cannot find `uint16_t` and other similar types by default anymore
+  env.CXXFLAGS = "-include cstdint";
 
   buildInputs = [
     hidapi
@@ -48,4 +51,4 @@ stdenv.mkDerivation rec {
     maintainers = with maintainers; [ fab ];
     platforms = platforms.linux;
   };
-}
+})

